test/ILLink.RoslynAnalyzer.Tests/AvoidAssemblyLocationInSingleFileTests.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = ILLink.RoslynAnalyzer.Tests.CSharpAnalyzerVerifier<
+	ILLink.RoslynAnalyzer.AvoidAssemblyLocationInSingleFile>;
+
+namespace ILLink.RoslynAnalyzer.Tests
+{
+	public class AvoidAssemblyLocationInSingleFileTests
+	{
+		[Fact]
+		public Task GetExecutingAssemblyLocation ()
+		{
+			const string src = @"
+using System.Reflection;
+class C
+{
+    public string M() => Assembly.GetExecutingAssembly().Location;
+}";
+
+			return VerifyCS.VerifyAnalyzerAsync (src,
+				new (string, string)[] { ($"build_property.{MSBuildPropertyOptionNames.PublishSingleFile}", "true") },
+				VerifyCS.Diagnostic (AvoidAssemblyLocationInSingleFile.IL3000).WithSpan (5, 26, 5, 66).WithArguments ("System.Reflection.Assembly.Location"));
```

Comment:
You are missing the comment here:
// (5,26): warning IL3000: 'System.Reflection.Assembly.Location' always returns an empty string for assemblies embedded in a single-file app. If the path to the app directory is needed, consider calling 'System.AppContext.BaseDirectory'.